src/dvclive/live.py

```diff
@@ -880,9 +883,19 @@ def make_dvcyaml(self):
         """
         make_dvcyaml(self)
 
-    @catch_and_warn(DvcException, logger)
-    def post_to_studio(self, event: Literal["start", "data", "done"]):
-        post_to_studio(self, event)
+    def post_data_to_studio(self):
+        if not self._studio_queue:
+            self._studio_queue = queue.Queue()
+
+            def worker():
+                while True:
+                    item = self._studio_queue.get()
+                    post_to_studio(item, "data")
+                    self._studio_queue.task_done()
+
+            threading.Thread(target=worker, daemon=True).start()
```

Comment:
where is post_to_studio defined now? 

One thing to test - Ctrl-C handling for this. Just to make sure that our threads and queue wait doesn't mess up apps in any way. Thanks @dberenbaum for iterating on this!

Reply:
should we wrap the call to post_to_studio in @catch_and_warn(DvcException, logger) ?

Reply:
> where is post_to_studio defined now?

https://github.com/iterative/dvclive/blob/edb5ee3182f93d3f5e0d5f63416ec83ea3daf431/src/dvclive/studio.py#L99

> One thing to test - Ctrl-C handling for this. Just to make sure that our threads and queue wait doesn't mess up apps in any way. Thanks @dberenbaum for iterating on this!

No, thanks for keeping me honest here. It's a pretty simple change but could easily see it breaking some things. This is one reason I wanted to keep start and done separate per your question above. The queue is set up in daemon mode which can exit without making all expected calls (without it I noticed tests were hanging). By separating start and end, we ensure those are called and returned in most scenarios. If a particular live data call gets missed, I don't think it's as critical.